File: notion-proxy.js
// Notion API proxy server - runs on port 3002
// Avoids CORS issues and keeps API key server-side
const http = require('http');
const https = require('https');

const PORT = 3002;
const NOTION_API = 'https://api.notion.com';
const NOTION_VERSION = '2022-06-28';
const API_KEY = 'ntn_I4762409521S2jhhbHClTBWTOX2rpu0MeW0SBAm7YmCaAW';

const server = http.createServer((req, res) => {
    // CORS headers
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

    if (req.method === 'OPTIONS') {
        res.writeHead(200);
        res.end();
        return;
    }

    // Route: /blocks/:id/children
    const blocksMatch = req.url.match(/^\/blocks\/([^/]+)\/children/);
    // Route: /pages/:id
    const pagesMatch = req.url.match(/^\/pages\/([^/?]+)/);

    let notionPath;
    if (blocksMatch) {
        notionPath = `/v1/blocks/${blocksMatch[1]}/children?page_size=100`;
    } else if (pagesMatch) {
        notionPath = `/v1/pages/${pagesMatch[1]}`;
    } else {
        res.writeHead(404, { 'Content-Type': 'application/json' });
        res.end(JSON.stringify({ error: 'Not found' }));
        return;
    }

    const options = {
        hostname: 'api.notion.com',
        path: notionPath,
        method: 'GET',
        headers: {
            'Authorization': `Bearer ${API_KEY}`,
            'Notion-Version': NOTION_VERSION,
            'Content-Type': 'application/json'
        }
    };

    const proxyReq = https.request(options, (proxyRes) => {
        let body = '';
        proxyRes.on('data', chunk => body += chunk);
        proxyRes.on('end', () => {
            res.writeHead(proxyRes.statusCode, { 'Content-Type': 'application/json' });
            res.end(body);
        });
    });

    proxyReq.on('error', (err) => {
        res.writeHead(500, { 'Content-Type': 'application/json' });
        res.end(JSON.stringify({ error: err.message }));
    });

    proxyReq.end();
});

server.listen(PORT, () => {
    console.log(`Notion proxy running on http://localhost:${PORT}`);
});

File: notion-proxy.py
#!/usr/bin/env python3
"""Notion API proxy server - runs on port 3002"""
import http.server
import json
import re
import ssl
import urllib.request

PORT = 3002
NOTION_API = 'https://api.notion.com'
NOTION_VERSION = '2022-06-28'
API_KEY = 'ntn_I4762409521S2jhhbHClTBWTOX2rpu0MeW0SBAm7YmCaAW'

class NotionProxy(http.server.BaseHTTPRequestHandler):
    def do_OPTIONS(self):
        self.send_response(200)
        self._cors()
        self.end_headers()

    def do_GET(self):
        blocks_match = re.match(r'^/blocks/([^/]+)/children', self.path)
        pages_match = re.match(r'^/pages/([^/?]+)', self.path)

        if blocks_match:
            notion_path = f'/v1/blocks/{blocks_match.group(1)}/children?page_size=100'
        elif pages_match:
            notion_path = f'/v1/pages/{pages_match.group(1)}'
        else:
            self.send_response(404)
            self._cors()
            self.end_headers()
            self.wfile.write(json.dumps({'error': 'Not found'}).encode())
            return

        try:
            req = urllib.request.Request(
                f'{NOTION_API}{notion_path}',
                headers={
                    'Authorization': f'Bearer {API_KEY}',
                    'Notion-Version': NOTION_VERSION,
                    'Content-Type': 'application/json'
                }
            )
            ctx = ssl.create_default_context()
            with urllib.request.urlopen(req, context=ctx) as resp:
                body = resp.read()
                self.send_response(resp.status)
                self._cors()
                self.send_header('Content-Type', 'application/json')
                self.end_headers()
                self.wfile.write(body)
        except urllib.error.HTTPError as e:
            body = e.read()
            self.send_response(e.code)
            self._cors()
            self.send_header('Content-Type', 'application/json')
            self.end_headers()
            self.wfile.write(body)
        except Exception as e:
            self.send_response(500)
            self._cors()
            self.end_headers()
            self.wfile.write(json.dumps({'error': str(e)}).encode())

    def _cors(self):
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')

    def log_message(self, format, *args):
        pass  # Suppress request logs

if __name__ == '__main__':
    server = http.server.HTTPServer(('', PORT), NotionProxy)
    print(f'Notion proxy running on http://localhost:{PORT}')
    server.serve_forever()
